### ISSUE TITLE
audio buffer has all zero's 

### ISSUE THREAD

**Issue by gpfreund** [saebekassebil/microphone-stream#10]
I'm trying to implement a audio recorder in React/Chrome (Windows 10). However the buffers show a zero. Below is fragment of the code. Any ideas would be very much appreciated:
```
var MicrophoneStream = require('microphone-stream');
var getUserMedia = require('get-user-media-promise');

    getUserMedia({video: false, audio: true})
      .then(function(mediaStream) {
        debugger;

        this.micStream = new MicrophoneStream(mediaStream, {/*objectMode: true*/});
        this.micStream.on('data', function(chunk) {
            switch (this.props.command) {
              case "start":
              case "resume": 
                this.onDataAvailable(chunk);
                break;
              case "stop":
                this.onStop();
            }
          }.bind(this)
        );
        
        this.onDataAvailable = function (chunk) {
          if (chunk.length > 0) {
            var wavBuf = new Int16Array(chunk.buffer);
            if (wavBuf.length > 0) {
              // this is just for testing - the buffer is all zeros
              var n;
              for (n = 0; n < wavBuf.length; n++) {
                if (wavBuf[n] != 0) {
                  console.log('Audio data received');
                  break;
                }
              }
            }            
          }
        }.bind(this);
        /*...*/
      }.bind(this)
    )
  ```

**Comment by nfriedly** [comment]
Hi @gpfreund,

I'm not sure what's going on here, and I'm out on vacation with just my phone right now. You might have to remind me next week if you don't hear back from me or anyone else by Tuesday or Wednesday.

**Comment by gpfreund** [comment]
Thanks, enjoy your vacation. I'll let you know if I figure it out in the meantime...
~G

**Comment by marcjoha** [comment]
Same issue on Chrome on Mac

**Comment by nfriedly** [comment]
A couple of questions: 

* Do you have a `.catch()` on the promise, and is it triggered?
* Does the browser give you a permissions dialog asking for microphone access? 
* Is you url either https or localhost? (See https://www.chromium.org/Home/chromium-security/prefer-secure-origins-for-powerful-new-features - we should probably link to that in the readme...)

**Comment by nfriedly** [comment]
Actually, I think the issue was due to `new Buffer()` being [deprecated](https://nodejs.org/api/buffer.html#buffer_new_buffer_array), followed by a small behavior change in the way it handles passed in Float32Arrays (previously it took the underlying data without modification, now it apparently loops over it and converts each value. Everything gets rounded to 0 in that process.)

I fixed it and added a couple of new tests. Should be released as v3.0.7 once the CI loop finishes.

Please try it out and confirm that it's working now.

**Comment by marcjoha** [comment]
That did the trick! Thanks @nfriedly :)
